File: go/MongoReplicaSet.go
// Assumes you have run 'go get gopkg.in/mgo.v2
package main

import (
    "fmt"
    "gopkg.in/mgo.v2"
)
//TODO: This hangs with the host is bogus. Is there a way to make it timeout?
func main() {
    Host := []string{
        "iad-c11-1.objectrocket.com:48152",
        "iad-c11-0.objectrocket.com:48152",
    }
    const (
        Username = "sooz"
        Password = "xxxxxxxx"
        Database = "test_db"
    )

    session, err := mgo.DialWithInfo(&mgo.DialInfo{
        Addrs:    Host,
        Username: Username,
        Password: Password,
        Database: Database,
    })
    if err != nil {
        panic(err)
    }

    fmt.Printf("Connected to replica set %v!\n", session.LiveServers())
}

// https://objectrocket.com/docs/mongodb_go_examples.html#connecting-to-a-replica-set

File: ruby/mongo_replica_set.rb
# Example Documentation: https://objectrocket.com/docs/mongodb_ruby_examples.html
# Driver Documentation: https://docs.mongodb.com/ruby-driver/master/

# Connecting to a replica set:
# The client has auto-discovery that will find all members of the replica set
# if not all are provided.

# !/usr/bin/env ruby
require 'mongo'

# Turn off debug-mode
Mongo::Logger.logger.level = Logger::WARN

client_host = ['iad-c11-1.objectrocket.com:48152,iad-c11-0.objectrocket.com:48152']
client_options = {
  database: 'test_db',
  replica_set: '99fc32e3c05e03597a692c4fd9a9d162',
  user: 'sooz',
  password: 'xxxxxxxx'
}

client = Mongo::Client.new(client_host, client_options)

puts('Client Connection: ')
puts(client.cluster.inspect)
puts
puts('Collection Names: ')
puts(client.database.collection_names)
